src/mower_logic/src/mower_logic/behaviors/UndockingBehavior.cpp

```diff
@@ -47,16 +51,41 @@ Behavior *UndockingBehavior::execute() {
 
   nav_msgs::Path path;
 
+  geometry_msgs::PoseStamped docking_pose_stamped_front;
+  docking_pose_stamped_front.pose = pose.pose.pose;
+  docking_pose_stamped_front.header = pose.header;
+
   int undock_point_count = config.undock_distance * 10.0;
   for (int i = 0; i < undock_point_count; i++) {
-    geometry_msgs::PoseStamped docking_pose_stamped_front;
-    docking_pose_stamped_front.pose = pose.pose.pose;
-    docking_pose_stamped_front.header = pose.header;
-    docking_pose_stamped_front.pose.position.x -= cos(yaw) * (i / 10.0);
-    docking_pose_stamped_front.pose.position.y -= sin(yaw) * (i / 10.0);
+    docking_pose_stamped_front.pose.position.x -= cos(yaw) * 0.1;
+    docking_pose_stamped_front.pose.position.y -= sin(yaw) * 0.1;
     path.poses.push_back(docking_pose_stamped_front);
   }
 
+  double angle;
+  if (config.undock_fixed_angle)
+    angle = config.undock_angle * (M_PI + M_PI) / 360.0;
+  else
+    angle = next_undock_angle;
```

Comment:
How about this?
```c++
double angle = config.undock_fixed_angle ? config.undock_angle * M_PI / 180.0 : next_undock_angle;
```

Reply:
With the random angle feature the `if` isn't as simple anymore, so I have not applied this change.